letta/server/server.py

```diff
@@ -1921,7 +1921,10 @@ def get_user_or_default(self, user_id: Optional[str]) -> User:
         if user_id is None:
             return self.get_default_user()
         else:
-            return self.get_user(user_id=user_id)
+            try:
+                return self.get_user(user_id=user_id)
+            except ValueError:
+                raise HTTPException(status_code=404, detail=f"User with id {user_id} not found")
```

Comment:
New behavior: if you include `user_id` in the headers but the `user_id` doesn't exist, we throw a 404 (not found).

We could alternatively throw an auth error, but it seems 404 is more appropricate?